### PR TITLE
[Catalog Promotion][UI] Fixed error when trying to save catalog promotion with empty percentage discount value

### PR DESCRIPTION
**Pull request by ernestWarwas**
| Q               | A
| --------------- | -----
| Branch?         | 1.11 
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no

When trying to edit the percentage discount value and save it with an empty amount there was a 500 error. This PR provides only a workaround for this problem.

image

image

### CODE REVIEW

File: src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php [hunk 1]
```diff
@@ -72,6 +73,12 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                         }
                     }
 
+                    if ($data['type'] === PercentageDiscountPriceCalculator::TYPE) {
+                        if ($data['configuration']['amount'] === '') {
+                            return;
+                        }
+                    }
+
```
Comment (AdamKasp) [inline src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php:81]: do we really want to hardcode these values (line 76 and 68), it looks like a workaround :(
what will happen with a new custom calculator?
I don't know any idea how to improve it :D but maybe @GSadee @Zales0123 or @lchrusciel  have some hint :)

Comment (lchrusciel) [reply src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php:81]: It should be cover with the regular validation and proper validation message

**Review by Zales0123 — APPROVED** [review]
This is definitely a workaround/hotfix and we need a deeper look into the problem... but it at least solves an issue and still can be fixed after the stable release 🖖

**Comment by Zales0123** [comment]
Thanks, @ernestWarwas! :tada:

**Pull request merged**